src/utils/createRequestHeaders.js

```diff
@@ -1,5 +1,9 @@
-export default async ({ getAccessToken, version } = {}) => ({
-  'Content-Type': 'application/json',
-  ...(getAccessToken && { Authorization: `Bearer ${await getAccessToken()}` }),
-  ...(version && { 'X-Api-Version': version }),
-});
+export default async ({ getAccessToken, version } = {}) => {
+  let accessToken = undefined;
+  return {
+    'Content-Type': 'application/json',
+    ...(getAccessToken &&
+      (accessToken = await getAccessToken()) !== undefined && { Authorization: `Bearer ${accessToken}` }),
```

Comment:
Use `typeof getAccessToken() !== 'undefined'` here and just use `let accessToken;` when initializing

Reply:
What do you mean by this? Do the `getAccessToken && (await getAccessToken())` directly in the let?

Reply:
My remark isn't relevant anymore after your changes 😄 looks good to me